src/poetry/core/factory.py

```diff
@@ -366,6 +367,10 @@ def create_dependency(
                     )
             else:
                 marker = parse_marker(markers)
+                # possibility that python constraint appears in markers, not keywords
+                if not python_versions and markers:
+                    python_versions_marker = get_python_constraint_from_marker(marker)
+                    python_versions = python_versions_marker
```

Comment:
The result of this parse is never used, is it?

The test that you have added passes without the fix, which is consistent with this not doing anything.  

Can you add a test that fails before the fix, and passes after it?